addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -855,7 +888,7 @@ public JsonBluetoothStates getBluetoothConnectionStates() {
     public JsonPlaylists getPlaylists(Device device) throws IOException, URISyntaxException {
         String json = makeRequestAndReturnString(
                 alexaServer + "/api/cloudplayer/playlists?deviceSerialNumber=" + device.serialNumber + "&deviceType="
-                        + device.deviceType + "&mediaOwnerCustomerId=" + device.deviceOwnerCustomerId);
+                        + device.deviceType + "&mediaOwnerCustomerId=" + this.accountCustomerId);
```

Comment:
You use the `this.accountCustomerId` in this case directly whereas you check for empty in lines [1120](https://github.com/openhab/openhab2-addons/pull/4385/files#diff-f33f578d83d597ddfea03dd4ae2241e4R1120) and [1192](https://github.com/openhab/openhab2-addons/pull/4385/files#diff-f33f578d83d597ddfea03dd4ae2241e4R1192). Which one is the safer way?

StringUtils.isEmpty(this.accountCustomerId) ? device.deviceOwnerCustomerId
                            : this.accountCustomerId)

Reply:
Your are right, it should be checked everywhere.